server/controllers/credit_card_controller.py

```diff
@@ -14,6 +14,15 @@ def get_credit_card_by_number(self, card_number):
         credit_card = CreditCard.query.filter_by(card_number=card_number).first()
         return jsonify(credit_card.to_dict() if credit_card else {})
 
+    def get_credit_card_balance(self, card_number):
```

Comment:
Meta thing: Per our style guide, can we add type hints for these inputs and also what's returned by the method(s).  For this function and others in this PR - will make it easier to understand